tooling/nargo_cli/src/cli/info_cmd.rs

```diff
@@ -273,13 +280,22 @@ fn count_opcodes_and_gates_in_program(
     package: &Package,
     expression_width: ExpressionWidth,
 ) -> Result<ProgramInfo, CliError> {
-    Ok(ProgramInfo {
-        name: package.name.to_string(),
-        expression_width,
-        // TODO(https://github.com/noir-lang/noir/issues/4428)
-        acir_opcodes: compiled_program.program.functions[0].opcodes.len(),
-        circuit_size: backend.get_exact_circuit_size(&compiled_program.program)?,
-    })
+    let functions = compiled_program
+        .program
+        .functions
+        .into_par_iter()
+        .enumerate()
+        .map(|(i, function)| -> Result<_, BackendError> {
+            Ok(FunctionInfo {
+                name: compiled_program.names[i].clone(),
+                acir_opcodes: function.opcodes.len(),
+                circuit_size: backend
+                    .get_exact_circuit_size(&Program { functions: vec![function] })?,
+            })
+        })
+        .collect::<Result<_, _>>()?;
```

Comment:
Not a blocker but this does scream that we should have the backend handle a program with multiple functions.

Reply:
Yes agreed. I forgot to note that. It would have required separate backend changes though, this PR allows us to at least look at the separate ACIR counts.

Reply:
Yeah, no worries. Not a blocker as this works and fixing this would require shifting the entire PR over to aztec-packages.